contracts/compounder/rewardHandlers/BalancerSwapsHandler.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { SafeERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/IERC20.sol";
+import { IBalancerVault, IAsset } from "../../interfaces/balancer/IBalancerCore.sol";
+import { IRewardHandler } from "../../interfaces/balancer/IRewardHandler.sol";
+import { HandlerBase } from "./HandlerBase.sol";
+
+contract BalancerSwapsHandler is HandlerBase {
+    using SafeERC20 for IERC20;
+
+    // ----------------------------------------------------------------
+    // Types
+    // ----------------------------------------------------------------
+
+    struct SwapPath {
+        bytes32[] poolIds;
+        address[] assetsIn;
+    }
+    address public immutable AURA_TOKEN;
+    address public immutable AURABAL_TOKEN;
+    SwapPath internal swapPath;
+
+    constructor(
+        address _token,
+        address _strategy,
+        address _balancerVault,
+        address _weth,
+        address _aura,
+        address _auraBal,
+        SwapPath memory _swapPath
+    ) HandlerBase(_token, _strategy, _balancerVault, _weth) {
+        AURA_TOKEN = _aura;
+        AURABAL_TOKEN = _auraBal;
+        _validateSwapPath(_swapPath, _token, _aura, _auraBal);
+        swapPath = _swapPath;
+    }
+
+    function _validateSwapPath(
+        SwapPath memory _swapPath,
+        address _token,
+        address _aura,
```

Comment:
Unused var? Seems like it can be removed. Same for `_auraBal` and actually because the swap path is immutable we don't really need to validate it on chain so we can just remove this function.